lib/cea/cea608_data_channel.js

```diff
@@ -0,0 +1,720 @@
+   * @param {!number} b1 Byte 1.
+   * @param {!number} b2 Byte 2.
+   * @return {!boolean}
+   * @private
+   */
+  isBackgroundAttribute_(b1, b2) {
+    // For Background Attribute Codes, the bytes take the following form:
+    // Bg provided: b1 -> |0|0|0|1|C|0|0|0| b2 -> |0|0|1|0|COLOR|T|
+    // No Bg:       b1 -> |0|0|0|1|C|1|1|1| b2 -> |0|0|1|0|1|1|0|1|
+    return (((b1 & 0xf7) === 0x10) && ((b2 & 0xf0) === 0x20)) ||
+             (((b1 & 0xf7) === 0x17) && ((b2 & 0xff) === 0x2D));
+  }
+
+  /**
+   * Checks if the character is in the Special North American char. set.
+   * @param {!number} b1 Byte 1.
+   * @param {!number} b2 Byte 2.
+   * @return {!boolean}
+   * @private
+   */
+  isSpecialNorthAmericanChar_(b1, b2) {
+    // The bytes take the following form:
+    // b1 -> |0|0|0|1|C|0|0|1|
+    // b2 -> |0|0|1|1|  CHAR |
+    return ((b1 & 0xf7) === 0x11) && ((b2 & 0xf0) === 0x30);
+  }
+
+  /**
+   * Checks if the character is in the Extended Western European char. set.
+   * @param {!number} b1 Byte 1.
+   * @param {!number} b2 Byte 2.
+   * @return {!boolean}
+   * @private
+   */
+  isExtendedWesternEuropeanChar_(b1, b2) {
+    // The bytes take the following form:
+    // b1 -> |0|0|0|1|C|0|1|S|
+    // b2 -> |0|0|1|CHARACTER|
+    return ((b1 & 0xf6) === 0x12) && ((b2 & 0xe0) === 0x20);
+  }
+
+  /**
+   * Checks if the data contains a control code.
+   * @param {!number} b1 Byte 1.
+   * @return {!boolean}
+   */
+  static isControlCode(b1) {
+    // For control codes, the first byte takes the following form:
+    // b1 -> |P|0|0|1|X|X|X|X|
+    return (b1 & 0x70) === 0x10;
+  }
+};
+
+/**
+ * Command codes.
+ * @enum {!number}
+ * @private
+ */
+shaka.cea.Cea608DataChannel.MiscCmd_ = {
+  // "RCL - Resume Caption Loading"
+  RCL: 0x20,
+
+  // "BS  - BackSpace"
+  BS: 0x21,
+
+  // "AOD - Unused (alarm off)"
+  AOD: 0x22,
+
+  // "AON - Unused (alarm on)"
+  AON: 0x23,
+
+  // "DER - Delete to End of Row"
+  DER: 0x24,
+
+  // "RU2 - Roll-Up, 2 rows"
+  RU2: 0x25,
+
+  // "RU3 - Roll-Up, 3 rows"
+  RU3: 0x26,
+
+  // "RU4 - Roll-Up, 4 rows"
+  RU4: 0x27,
+
+  // "FON - Flash On"
+  FON: 0x28,
+
+  // "RDC - Resume Direct Captions"
+  RDC: 0x29,
+
+  // "TR - Text Restart"
+  TR: 0x2a,
+
+  // "RTD - Resume Text Display"
+  RTD: 0x2b,
+
+  // "EDM - Erase Displayed Mem"
+  EDM: 0x2c,
+
+  // "CR  - Carriage return"
+  CR: 0x2d,
+
+  // "ENM - Erase Nondisplayed Mem"
+  ENM: 0x2e,
+
+  // "EOC - End Of Caption (flip mem)"
+  EOC: 0x2f,
+};
+
+/**
+ * Caption type.
+ * @private @const @enum {!number}
+ */
+shaka.cea.Cea608DataChannel.CaptionType = {
+  NONE: 0,
+  POPON: 1,
+  PAINTON: 2,  // From here on are all "painty" (painton)
+  ROLLUP: 3,   // From here on are all "rolly" (rollup)
+  TEXT: 4,
+};
+
+/**
+ * 608 closed captions data - time, byte 1, byte 2, and the order in which the
+ * data was received.
+ * @typedef {{
+ *   pts: number,
+ *   type: number,
+ *   cc1: number,
```

Comment:
Are these the two bytes that are referred to as "b1" and "b2" elsewhere?  Since "CC1" is the name of a closed caption stream, it might be confusing to name this "cc1" if there's a better name we could give it.

Reply:
Changed the name to what the spec officially calls them. (ccData1 and ccData2).